In message <20030223.214513.120185268.davem@redhat.com>,"David S. Miller" writes:
>Don't try to modify skb->{data,len} by hands, let the skb_*()
>interfaces do it.  Use skb_pull() in this case.

missed that function when i went looking for it.  again, the right way:

Index: linux/net/atm/lec.c
===================================================================
RCS file: /home/chas/CVSROOT/linux/net/atm/lec.c,v
retrieving revision 1.1
retrieving revision 1.7
diff -u -d -b -w -r1.1 -r1.7
--- linux/net/atm/lec.c	20 Feb 2003 13:46:30 -0000	1.1
+++ linux/net/atm/lec.c	24 Feb 2003 13:34:43 -0000	1.7
@@ -711,7 +705,7 @@
                         lec_arp_check_empties(priv, vcc, skb);
                 }
                 skb->dev = dev;
-                skb->data += 2; /* skip lec_id */
+                skb_pull(skb, 2); /* skip lec_id */
 #ifdef CONFIG_TR
                 if (priv->is_trdev) skb->protocol = tr_type_trans(skb, dev);
                 else
-
To unsubscribe from this list: send the line "unsubscribe linux-kernel" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html
Please read the FAQ at  http://www.tux.org/lkml/
